addons/calendar/models/res_users.py

```diff
@@ -20,6 +28,52 @@ def SELF_READABLE_FIELDS(self):
     def SELF_WRITEABLE_FIELDS(self):
         return super().SELF_READABLE_FIELDS + ['calendar_default_privacy']
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        """ Set the calendar default privacy as the same as Default User Template when defined. """
+        # Get the calendar default privacy from the Default User Template, set public as default.
+        default_privacy = 'public'
+        default_user = self.env.ref('base.default_user', raise_if_not_found=False)
+        if default_user and default_user.calendar_default_privacy:
+            default_privacy = default_user.calendar_default_privacy
+
+        # Update the dictionaries in vals_list with the calendar default privacy.
+        for vals_dict in vals_list:
+            if not vals_dict.get('calendar_default_privacy'):
+                vals_dict.update(calendar_default_privacy=default_privacy)
+
+        res = super().create(vals_list)
+        return res
+
+    def write(self, vals):
+        """ Forbid the calendar default privacy update from different users for keeping private events secured. """
+        privacy_update = 'calendar_default_privacy' in vals
+        default_user = self.env.ref('base.default_user', raise_if_not_found=False)
+        if default_user and privacy_update and any(user not in [default_user, self.env.user] for user in self):
+            raise AccessError(_("You are not allowed to change the calendar default privacy of another user due to privacy constraints."))
```

Comment:
Admin/Odoo bot is not authorized to update the default setting. Shoudn't be possible for odoobot to do it ? TBD.